### DISCUSSION TITLE
SQLite + IndexedDB => Absurd-SQL

### DISCUSSION THREAD

**Discussion by edlaver** [Nozbe/WatermelonDB#1139] [General]
@radex Thought you might be interested in this, an implementation of SQLite in the browser that maintains persistence via IndexedDB:

https://jlongster.com/future-sql-web

https://github.com/jlongster/absurd-sql

> It implements a backend for sql.js (sqlite3 compiled for the web) that treats IndexedDB like a disk and stores data in blocks there. That means your sqlite3 database is persisted. And not in the terrible way of reading and writing the whole image at once -- it reads and writes your db in small chunks.

**Comment by radex** [comment]
See: https://github.com/Nozbe/WatermelonDB/discussions/1545
